libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -147,43 +157,32 @@ Compass::accept_calibration(uint8_t i)
 }
 
 bool
-Compass::accept_calibration_mask(uint8_t mask)
+Compass::_accept_calibration_mask(uint8_t mask)
 {
-    for(uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
-        if ((1<<i) & mask) {
-            CompassCalibrator& cal = _calibrator[i];
-            uint8_t cal_status = cal.get_status();
-            if (cal_status != COMPASS_CAL_SUCCESS && cal_status != COMPASS_CAL_NOT_STARTED) {
-                // a compass failed or is still in progress
-                return false;
-            }
-        }
-    }
-
     bool success = true;
     for (uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
         if ((1<<i) & mask) {
-            if (!accept_calibration(i)) {
+            if (!_accept_calibration(i)) {
                 success = false;
             }
+            _calibrator[i].clear();
```

Comment:
Shouldn't this be inside an else? Otherwise you can be canceling a calibration that you were told (wrongly) to accept.